src/AmazonOrderList.php

```diff
@@ -469,15 +470,17 @@ protected function parseXML($xml)
             if ($key != 'Order') {
                 break;
             }
-            $this->orderList[$this->index] = new AmazonOrder(
+
+            $order = new AmazonOrder(
                 $this->storeName,
                 null,
                 $data,
                 $this->mockMode,
                 $this->mockFiles,
                 $this->config
             );
-            $this->orderList[$this->index]->mockIndex = $this->mockIndex;
```

Comment:
This line needs to stay for tests I believe.

Reply:
I was thinking the same. Unfortunately leaving it in as it makes the code fail and request not passing on. Couldn't figure out if where the `$this->mockIndex` is coming from, seems like non-exists property. 

Any thoughts?

Reply:
Well, it's setting the mockIndex on the AmazonOrder object...I'm wondering if a better approach here might be to leave this as is, and you should be able to do something like this:

```
$order_list = new AmazonOrderList(....);
...
$order_list->getList();
foreach($order_list as $order) {
    $order_data = $order->getData();
}
```

Could you try that and let me know if that works?

Reply:
And BTW, `mockIndex` is a property on `AmazonCore`.

Reply:
You're right. it's actually working fine when putting it back - I think it's something with config although not really sure about it. anyway, looks like it working